waiter/src/waiter/scheduler/kubernetes.clj

```diff
@@ -248,21 +248,23 @@
 
 (defn pod->ServiceInstance
   "Convert a Kubernetes Pod JSON response into a Waiter Service Instance record."
-  [{:keys [api-server-url] :as scheduler} pod]
+  [{:keys [api-server-url proxy-options] :as scheduler} pod]
   (try
     (let [;; waiter-app is the first container we register
           restart-count (get-in pod [:status :containerStatuses 0 :restartCount] 0)
           instance-id (pod->instance-id pod restart-count)
           node-name (get-in pod [:spec :nodeName])
-          port0 (get-in pod [:spec :containers 0 :ports 0 :containerPort])
+          pod-annotations (get-in pod [:metadata :annotations])
+          has-reverse-proxy? (contains? pod-annotations :waiter/reverse-proxy)
+          offset (if has-reverse-proxy? (:reverse-proxy-offset proxy-options) 0)
```

Comment:
Earlier, we spoke about reading the base port from the pod annotation. 
Then we no longer need access to `proxy-options` or the need to compute offset.

```
port0 (or (some-> (get-in pod [:metadata :annotations :waiter/base-port]) (Integer/parseInt))
          (get-in pod [:spec :containers 0 :ports 0 :containerPort]))
```